lib/src/slices/data_bkg/data_bkg_service.dart

```diff
@@ -13,265 +12,156 @@ import '../api_app_data/model/api_app_data_model.dart';
-    ApiAppDataModel? appDataGmailPage =
-        await _apiAppDataService.getByKey(ApiAppDataKey.gmailPage);
-    page = appDataGmailPage?.value;
-    return _checkGmailFetchListPage(query: query, page: page);
+        this._apiEmailMsgService = apiEmailMsgService {
+    getExistingAccounts().then((_) => fetchAllData());
   }
 
-  Future<void> _checkGmailFetchListPage({String? page, String? query}) async {
-    DataBkgModelPage<String> res = await _apiGoogleService.gmailFetch(
-        query: query, page: page, maxResults: 5);
-    if (res.data != null) {
-      List<String> known =
-          (await _apiEmailMsgService.getByExtMessageIds(res.data!))
-              .map((message) => message.extMessageId!)
-              .toList();
-      List<String> unknown =
-          res.data!.where((message) => !known.contains(message)).toList();
-      await _checkGmailFetchMessage(unknown);
-    }
-    await _apiAppDataService.save(ApiAppDataKey.gmailPage, res.next ?? '');
-    if (res.next != null)
-      return _checkGmailFetchListPage(page: res.next, query: query);
+  Future<void> getExistingAccounts() async {
+    _accounts = await _apiAuthService.getAllAccounts();
+    notifyListeners();
   }
 
-  Future<void> _checkGmailFetchMessage(List<String> messages) async {
-    Set<String> processed = Set();
-    for (String messageId in messages) {
-      ApiEmailMsgModel? message = await _apiGoogleService.gmailFetchMessage(
-          messageId,
-          format: "metadata",
-          headers: ["List-unsubscribe"]);
-      if (message?.extMessageId != null) processed.add(message!.extMessageId!);
-      if (message?.sender?.email != null &&
-          message?.sender?.unsubscribeMailTo != null) {
-        ApiEmailSenderModel? sender =
-            await _apiEmailSenderService.getByEmail(message!.sender!.email!);
-        if (sender != null) {
-          await _saveSender(sender);
-          await _apiEmailMsgService.upsert(message);
-          _log.fine('Sender upsert: ' + (sender.company?.domain ?? ''));
-          notifyListeners();
-        } else {
-          Set<ApiEmailMsgModel> senderMessages =
-              await _checkGmailNewSender(message.sender!.email!);
-          Set<String> senderMessageIds =
-              senderMessages.map((message) => message.extMessageId!).toSet();
-          List<String> newMessages = messages
-              .where((message) =>
-                  !senderMessageIds.contains(message) &&
-                  !processed.contains(message))
-              .toList();
-          return _checkGmailFetchMessage(newMessages);
-        }
-      }
+  Future<void> fetchAllData() async {
+    _log.fine("fetch all data started");
+    int startIndex = await _getStartIndex();
+    for (int i = startIndex; i < _accounts.length; i++) {
+      ApiAuthServiceAccountModel account = _accounts[i];
+      await fetchData(account);
+      await _apiAppDataService.save(
+          ApiAppDataKey.dataBkgLastAccount, i.toString());
```

Comment:
You can't guarantee order. We need to store the account, the service (email), and specifics for the service (category, and page)